js/src/utils.ts

```diff
@@ -0,0 +1,56 @@
+import { PublicKey } from '@solana/web3.js';
+import type BN from 'bn.js';
+import type { ProgramAddresses, Snapshot } from './types';
+
+export const findAuthorityProgramAddress = async (
+  programAddresses: ProgramAddresses,
+  additionalSeedString: string
```

Comment:
Can this be more strictly typed? AFAICT the only relevant strings are `reserve_account` and `mint_authority`. Are `rewards_withdraw_authority` and `sol_reserve` also valid?

Reply:
Yeah good idea, we can add all the available options to it like "reserve_account" | "mint_authority" ...
To help the user(editor suggestions) and to make it more strictly typed